enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,539 @@
+---
+title: assisted-installer-in-cluster
+authors:
+  - "@mhrivnak"
+  - "@hardys"
+  - "@avishayt"
+reviewers:
+  - "@eparis"
+  - "@markmc"
+  - "@dgoodwin"
+  - "@ronniel1"
+
+approvers:
+  - TBD
+creation-date: 2020-12-22
+last-updated: 2021-02-09
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced|informational
+see-also:
+replaces:
+superseded-by:
+---
+
+# Assisted Installer in-cluster
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The Assisted Installer currently runs as a SaaS on cloud.redhat.com, enabling
+users to deploy OpenShift clusters with certain customizations, particularly on
+bare metal hardware. It is necessary to bring those capabilities on-premise in
+users' "Hub" clusters by installing clusters via Multi-cluster management, such
+as through Hive and RHACM.
+
+This enhancement proposes the first iteration of running the Assisted Installer
+in end-user clusters to meet the purpose above.
+
+## Motivation
+
+### Goals
+
+* Expose the Assisted Installer's capabilities as a kubernetes-native API. Some
+portion may be exposed through [Hive](https://github.com/openshift/hive)'s API.
+* Enable multi-cluster management tooling to create new clusters using Assisted Installer.
+* Enable adding workers to any OpenShift cluster deployed by the Assisted Installer via
+  the same multi-cluster management tooling.
+* Enable automated creation and booting of the Assisted discovery ISO for bare-metal deployments.
+* Ensure the design is extensible for installing on other platforms.
+
+### Non-Goals
+
+* Remote Worker Node support requires solving specific problems that will be
+addressed in a separate proposal.
+* Solve central machine management. That can be done with this effort, or after
+this effort, but it is not strictly a requirement in order to deliver the
+goals stated above.
+* Run metal3 components on a non-baremetal cluster.
+
+## Proposal
+
+### User Stories
+
+#### Install cluster from hub
+
+As a user of Red Hat's multi-cluster mangement tools, I can use the assisted
+installation agent-based workflow to create clusters from a pool of bare metal
+inventory.
+
+#### Add Worker Node from Hub
+
+After deploying a cluster with Assisted Installer, I can add workers from bare
+metal hardware on day two using a similar workflow and tool set. I can either
+obtain the discovery live ISO and use my own methods to boot it, or I can use
+the baremetal-operator to boot the live ISO automatically.
```

Comment:
Is the goal to choose one workflow to implement, or to implement both? I think both? Calling them out as separate use cases would clarify that.

Reply:
The goal is both, and this actually applies to both cluster creation and adding workers on day 2. I'll try adding additional use cases for "Boot it Yourself" and "Automated booting", which are both ways to get Agents running.